src/openrct2/actions/StaffHireNewAction.hpp

```diff
@@ -54,13 +54,13 @@ DEFINE_GAME_ACTION(StaffHireNewAction, GAME_COMMAND_HIRE_NEW_STAFF_MEMBER, Staff
 {
 private:
     bool _autoPosition = false;
-    uint8_t _staffType = STAFF_TYPE::STAFF_TYPE_COUNT;
+    StaffType _staffType = StaffType::Count;
```

Comment:
See comments on discord. Keep this a uint8_t but keep the constructor with StaffType for the parameter. Cast it in the body of the constructor.